### PR TITLE
Add querystring support.

### PR DESCRIPTION
**Pull request by ericelliott**
Adds a new `queryName` option to customize the name of the query used to change the language.

### CODE REVIEW

**Comment by ericelliott** [comment]
Oops. Looks like the tests failed in CI. I'll take a look and figure out why.

**Comment by ericelliott** [comment]
Looks like it's configured for old versions of Node without `Object.assign()` support. Fixed. =)

File: index.js [hunk 1]
```diff
@@ -71,6 +71,16 @@ module.exports = function(props) {
 
   return function(req, res, next) {
     var language;
+    var queryName = props.queryName || 'language';
+    var queryLanguage = req.query[queryName];
+
+    if(typeof queryLanguage === 'string' && queryLanguage.length > 1) {
```
Comment (tinganho) [inline index.js]: Can you add a check that the language is supported too? `props.languages.indexOf(queryLanguage) !== -1`.

Also add a corresponding test for it.

File: test/test.js [hunk 2]
```diff
@@ -255,4 +264,41 @@ describe('request-language', function() {
...
+    });
+    var req = getRequest({
+      acceptLanguage: 'en-US;q=1',
+      query: {
+        language: 'zh-CN'
+      }
+    });
+
+    middleware(req, res, next);
+    expect(req.cookies.language).to.equal('zh-CN');
```
Comment (tinganho) [inline test/test.js:283]: Can you also check for `expect(req.language).to.equal('zh-CN');` ?

File: test/test.js [hunk 3]
```diff
@@ -255,4 +264,41 @@ describe('request-language', function() {
...
+    });
+    var req = getRequest({
+      acceptLanguage: 'en-US;q=1',
+      query: {
+        locale: 'zh-CN'
+      }
+    });
+
+    middleware(req, res, next);
+    expect(req.cookies.language).to.equal('zh-CN');
```
Comment (tinganho) [inline test/test.js:322]: Same here.

File: test/test.js [hunk 4]
```diff
@@ -255,4 +264,41 @@ describe('request-language', function() {
...
+  });
```
Comment (tinganho) [inline test/test.js:324]: Please also add a before mentioned test if the language is not supported. It should continue the normal logic chain(check cookie and if not check accept-lang header.).

File: README.md [hunk 5]
```diff
@@ -90,6 +90,19 @@ The options are the same options as express uses in `res.cookie(name, value. opt
 ##### cookie.url (optional)
 Set the change language URL. Lets say that you set the value to `/languages/{language}` in your configurations. If you visit with your browser the URL path `/languages/en-US`. It will change your language cookie value to `en-US`. It will redirect back to the origin URL if you send a referrer header and default to `/` if it don't send a referrer header.
 
+#### queryName (optional) \{Object\}
+You can optionally set the language using a query string. This option allows you to set the name of the query parameter that triggers the language setting.
```
Comment (tinganho) [inline README.md]: Can you add somewhere that the queryName defaults to `language`?

**Comment by tinganho** [comment]
Hi @ericelliott thanks for sending a PR. I left some comments it would be great if you could take a look at them. Otherwise I like the proposed API change.

**Comment by ericelliott** [comment]
Great feedback! Thanks for being so thorough in your review. This is what a good code review looks like!

(ﾉ◕ヮ◕)ﾉ*:･ﾟ✧

File: test/test.js (hunk 2 above)
Comment (guidsen) [inline test/test.js:283]: Check function requestLangauge for typo

File: test/test.js [hunk 6]
```diff
@@ -255,4 +264,41 @@ describe('request-language', function() {
...
+        language: 'zh-CN'
+      }
+    });
+
+    middleware(req, res, next);
+    expect(req.cookies.language).to.equal('zh-CN');
+  });
+
+  it('should be able to use a custom query string name', function() {
+    var middleware = requestLangauge({
```
Comment (guidsen) [inline test/test.js]: Functionname typo here as well

File: test/test.js (hunk 2 above)
Comment (ericelliott) [inline test/test.js:283]: What typo? Please supply expected vs actual.

File: test/test.js [hunk 7]
```diff
@@ -255,4 +264,41 @@ describe('request-language', function() {
     };
     expect(method).to.throw('You haven\'t defined the markup `{language}` in your cookie.url settings.');
   });
+
+  it('should set the language requested in the query string', function() {
+    var middleware = requestLangauge({
```
Comment (guidsen) [inline test/test.js]: Did you mean `requestLangauge` instead of `requestLanguage`

Comment (guidsen) [inline test/test.js]: I mean the other way around. So `requestLanguage` instead of `requestLangauge`

File: test/test.js (hunk 6 above)
Comment (guidsen) [inline test/test.js]: Did you mean `requestLanguage` instead of `requestLangauge`

File: test/test.js (hunk 7 above)
Comment (ericelliott) [inline test/test.js]: Oh, I see it now. I just copied and pasted existing examples, so I didn't even notice it was spelled wrong. It's spelled wrong in the import,  as well, which is why the code works anyway. =)

Comment (guidsen) [inline test/test.js]: Ah ok I might create a PR to fix the typo in some minutes.

Comment (ericelliott) [inline test/test.js]: Cool. Thanks. And good catch! =)

Comment (guidsen) [inline test/test.js]: Fixed the typo in #2

Comment (tinganho) [inline test/test.js]: Just merged #2. Thanks for noticing the typo from me.

**Comment by tinganho** [comment]
thanks @ericelliott

**Comment by ericelliott** [comment]
Fixed. =)

**Comment by tinganho** [comment]
Thanks @ericelliott

**Pull request merged**